examples/trait-module-plugin/create-simple-trait/model/custom-trait.smithy

```diff
@@ -0,0 +1,39 @@
+$version: "2.0"
+
+namespace io.smithy.gradle.example
+
+@trait(
+    selector: "resource"
+    breakingChanges: [
+        {
+            change: "presence"
+        }
+    ]
+)
+structure resourceMetadata {
+    /// Provides a custom name for your resource.
+    @required
+    description: String
+
+    /// A type for the resource
+    @required
+    type: ResourceType
+
+    /// A list of associated structures
+    associated: Associated
+}
+
+@private
+@idRef(failWhenMissing: true, selector: "structure")
+string Associated
+
+@private
+enum ResourceType {
+    NORMAL
+    SPECIAL
+    OTHER
+    NONE
+}
+
+@trait(selector: "member")
+string jsonName
```

Comment:
nit: IMO, it would be better to name this something else, since `jsonName` is already a commonly used trait.